fdbserver/workloads/SpecialKeySpaceCorrectness.actor.cpp

```diff
@@ -201,6 +208,120 @@ struct SpecialKeySpaceCorrectnessWorkload : TestWorkload {
 
 		return GetRangeLimits(rowLimits, byteLimits);
 	}
+
+	static void testConflictRanges(Database cx_, bool read) {
+		StringRef prefix = read ? readConflictRangeKeysRange.begin : writeConflictRangeKeysRange.begin;
+		TEST(read); // test read conflict range special key implementation
+		TEST(!read); // test write conflict range special key implementation
+		// Get a default special key range instance
+		Database cx = cx_->clone();
+		Reference<ReadYourWritesTransaction> tx = Reference(new ReadYourWritesTransaction(cx));
+		Reference<ReadYourWritesTransaction> referenceTx = Reference(new ReadYourWritesTransaction(cx));
+		bool ryw = deterministicRandom()->coinflip();
+		if (!ryw) {
+			tx->setOption(FDBTransactionOptions::READ_YOUR_WRITES_DISABLE);
+		}
+		referenceTx->setVersion(100); // Prevent this from doing a GRV or committing
+		referenceTx->clear(normalKeys);
+		referenceTx->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+		int numKeys = deterministicRandom()->randomInt(1, 10) * 4;
+		std::vector<std::string> keys; // Must all be distinct
+		keys.resize(numKeys);
+		int lastKey = 0;
+		for (auto& key : keys) {
+			key = std::to_string(lastKey++);
+		}
+		if (deterministicRandom()->coinflip()) {
+			// Include beginning of keyspace
+			keys.push_back("");
+		}
+		if (deterministicRandom()->coinflip()) {
+			// Include end of keyspace
+			keys.push_back("\xff");
+		}
+		std::mt19937 g(deterministicRandom()->randomUInt32());
+		std::shuffle(keys.begin(), keys.end(), g);
+		// First half of the keys will be ranges, the other keys will mix in some read boundaries that aren't range
+		// boundaries
+		std::sort(keys.begin(), keys.begin() + keys.size() / 2);
+		for (auto iter = keys.begin(); iter + 1 < keys.begin() + keys.size() / 2; iter += 2) {
+			Standalone<KeyRangeRef> range = KeyRangeRef(*iter, *(iter + 1));
+			if (read) {
+				tx->addReadConflictRange(range);
```

Comment:
Minor: I think the code already handled the case correctly when `readYourWritesDisabled=true`. 
But since the test cases here are all non-overlapping ranges, it is better to test overlapping cases as well.

Reply:
I'm basically trying to test that we're actually using a CoalescedKeyRefRangeMap and not just sorting the conflict ranges and returning those, since that would be incorrect.